### ISSUE TITLE
:meck throws MatchError - no match of right hand side value when used in other deps in Elixir 1.5

### ISSUE THREAD

**Issue by taher435** [eproxus/meck#182]
`:meck` is being used by [excoveralls](https://github.com/parroty/excoveralls) and [coverex](https://github.com/alfert/coverex). Both the deps throw the same error when we do

```
mix test coveralls
```

Elixir version: `1.5.0`

Below is the error stack strace

```
** (MatchError) no match of right hand side value: [error: {:no_abstract_code, <<70, 79, 82, 49, 0, 0, 3, 132, 66, 69, 65, 77, 65, 116, 85, 56, 0, 0, 1, 39, 0, 0, 0, 18, 52, 69, 108, 105, 120, 105, 114, 46, 80, 111, 108, 108, 105, 110, 97, 116, 111, 114, 46, 71, 111, 111, 100, 115, 104, ...>>}]
    /Users/taher/elixir-repo/deps/meck/src/meck_cover.erl:32: :meck_cover.compile_beam/2
    /Users/taher/elixir-repo/deps/meck/src/meck_proc.erl:387: :meck_proc.backup_original/3
    /Users/taher/elixir-repo/deps/meck/src/meck_proc.erl:206: :meck_proc.init/1
    (stdlib) gen_server.erl:365: :gen_server.init_it/2
    (stdlib) gen_server.erl:333: :gen_server.init_it/6
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:EXIT, #PID<0.73.0>, {{:badmatch, [error: {:no_abstract_code, <<70, 79, 82, 49, 0, 0, 3, 132, 66, 69, 65, 77, 65, 116, 85, 56, 0, 0, 1, 39, 0, 0, 0, 18, 52, 69, 108, 105, 120, 105, 114, 46, 80, 111, 108, 108, 105, 110, 97, 116, 111, 114, 46, 71, 111, ...>>}]}, [{:meck_cover, :compile_beam, 2, [file: '/Users/taher/elixir-repo/deps/meck/src/meck_cover.erl', line: 32]}, {:meck_proc, :backup_original, 3, [file: '/Users/taher/elixir-repo/deps/meck/src/meck_proc.erl', line: 387]}, {:meck_proc, :init, 1, [file: '/Users/taher/elixir-repo/deps/meck/src/meck_proc.erl', line: 206]}, {:gen_server, :init_it, 2, [file: 'gen_server.erl', line: 365]}, {:gen_server, :init_it, 6, [file: 'gen_server.erl', line: 333]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 247]}]}}
13:26:41.782 [error] Task #PID<0.1546.0> started from :error_logger terminating
```

## Reproduction Steps

1. Install coveralls or coverex in your elixir repo
2. Run `mix test --cover`
3. See the error output

## Expected behavior

`:meck` should not throw `** (MatchError) no match of right hand side value:`

## Observed behavior

See the stack trace above

## Versions

* **Meck version:** `0.8`
* **Erlang version:** OTP 20 (Elixir 1.5.0)

OS: MacOS Sierra
Installed Erlang using `brew install elixir` which installs the latest version of Elixir along with Erlang.

There is an issue reported in `excoveralls` which is related to this
https://github.com/parroty/excoveralls/issues/107

**Comment by eproxus** [comment]
Please try upgrading to the latest version of Meck which is 0.8.8 and see if the error persists.

**Comment by taher435** [comment]
Facing same issue. Updated the version to `0.8.8`
https://github.com/taher435/excoveralls/blob/master/mix.exs#L32

**Comment by eproxus** [comment]
From what I can gather from the error, it looks like `cover` cannot compile the beam files because they lack abstract code (being compiled with `debug_info`). Is this a possibility?

**Comment by ever-cifuentes-globant** [comment]
The error seems to be happening with Erlang 20.
I found the same error, but switching back to Erlang 19.3 with Kerl, the error went away.

**Comment by eproxus** [comment]
Not sure how to proceed here. Can anyone reproduce this with just Erlang 20 and Meck, or is it tied to Elixir/ExCoveralls/Coverex as well?

**Comment by eproxus** [comment]
@taher435 @ever-cifuentes-globant Do you still have this problem?

**Comment by edgurgel** [comment]
I had the same issue and using meck 0.8.8 fixed it. I was using `coverex` 1.4.13 and I didn't need to update it.

**Comment by eproxus** [comment]
@edgurgel Thanks for the input.

Closing this issue for now, but please re-open if it is not fixed by using 0.8.8 or other later versions!
